Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +12,197 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+                return "N/A.";
+
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            var biggestBalanceEver = float.MinValue;
+
+            foreach (var personBalances in peopleBalances)
+            {
+                foreach (var balance in personBalances.Value)
+                {
+                    if (balance > biggestBalanceEver)
+                    {
+                        names.Clear();
+                        biggestBalanceEver = balance;
+                        names.Add(personBalances.Key);
+                    }
+                    if (balance == biggestBalanceEver)
```

Comment:
Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5838869)